plugins/pagerduty/src/components/types.tsx

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright 2020 Spotify AB
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+export type Incident = {
+  id: string;
+  title: string;
+  status: string;
+  html_url: string;
```

Comment:
Since these types are used by the components throughout, I think it might be ideal to `camelCase` them (as per JS convention) and using a mapper/transform from the response to these.

Reply:
We had a discussion around this internally, and weren't really sure it was worth the tradeoff. The responses are quite a lot larger than the types listed so there would be quite a lot of mapping going on. You also run into some DX issues where it becomes a bit confusing to view the PD API docs and search SO etc.

Either way I think it could be sane to keep this all as plugin internal API for now by simply not exporting the `pagerDutyApiRef` from the plugin.

Reply:
Yeah, if they are just a pass through then it can additional work to create mappers. It's a practice I typically try to adhere to. The mapping functions albeit sometimes tedious, tend to provide a more closed system and remove coupling. That said, it's safe to make mapping optional assuming they are not exposed outside the package.

On an interesting note, this was added to TypeScript 4.1 https://github.com/microsoft/TypeScript/pull/40580, which opens the door for doing some interesting auto mapping.